src/zenml/services/container/container_service_endpoint.py

```diff
@@ -80,7 +80,7 @@ class ContainerServiceEndpoint(BaseServiceEndpoint):
     )
     monitor: Optional[
         Union[HTTPEndpointHealthMonitor, TCPEndpointHealthMonitor]
-    ] = Field(..., discriminator="type", union_mode="left_to_right")
+    ] = Field(..., discriminator="type")
```

Comment:
Why was the union mode removed moved here?

Reply:
@schustmi  We removed union_mode because Pydantic 2.9+ made it invalid to use union_mode on discriminated unions.

After 2.9: Pydantic raises RuntimeError: Unable to apply constraint 'union_mode' to schema of type 'tagged-union'\

We simply removed the redundant union_mode parameter: The discriminator="type" already tells Pydantic exactly how to choose the type